### PR TITLE
Exception for oversized plaintext

### PR DESCRIPTION
**Pull request by dsuponitskiy**

### CODE REVIEW

File: src/pke/include/encoding/plaintextfactory.h [hunk 1]
```diff
@@ -79,6 +79,15 @@ class PlaintextFactory {
     static Plaintext MakePlaintext(const std::vector<int64_t>& value, PlaintextEncodings encoding,
                                    std::shared_ptr<T> vp, EncodingParams ep, SCHEME schemeID = SCHEME::INVALID_SCHEME,
                                    size_t noiseScaleDeg = 1, uint32_t level = 0, NativeInteger scalingFactor = 1) {
+        // Check if plaintext has got enough slots for data (value)
+        usint ringDim = vp->GetRingDimension();
+        size_t valueSize = value.size();
+        if (SCHEME::CKKSRNS_SCHEME == schemeID && valueSize > ringDim / 2) {
+            OPENFHE_THROW(config_error, "The size [" + std::to_string(valueSize) + "] of the vector with values should not be greater than ringDim/2 [" + std::to_string(ringDim / 2) + "]");
```
Comment (sarojaduality) [inline src/pke/include/encoding/plaintextfactory.h:86]: I would add to the error message "for CKKS"

File: src/pke/include/encoding/plaintextfactory.h [hunk 2]
```diff
@@ -79,6 +79,15 @@ class PlaintextFactory {
...
+        }
+        else if (valueSize > ringDim) {
+            OPENFHE_THROW(config_error, "The size [" + std::to_string(valueSize) + "] of the vector with values should not be greater than ringDim [" + std::to_string(ringDim) + "]");
```
Comment (sarojaduality) [inline src/pke/include/encoding/plaintextfactory.h:89]: I would to the error message "for BGV/BFV schemes"

File: src/pke/include/cryptocontext.h [hunk 3]
```diff
@@ -124,11 +124,24 @@ class CryptoContextImpl : public Serializable {
             else {
                 elemParamsPtr = cryptoParams->GetElementParams();
             }
-
+            // Check if plaintext has got enough slots for data (value)
```
Comment (sarojaduality) [inline src/pke/include/cryptocontext.h:127]: Why do we need to add the CKKS plaintext check here if it's already in MakePlaintext?

**Review by sarojaduality — COMMENTED** [review]
Left a few questions.

File: src/pke/include/cryptocontext.h (hunk 3 above)
Comment (dsuponitskiy) [reply src/pke/include/cryptocontext.h:127]: Unfortunately we do not call MakePlaintext() here. I have a TODO comment added to change this

**Review by sarojaduality — APPROVED** [review]

**Pull request merged**